include/swift/AST/DiagnosticEngine.h

```diff
@@ -629,6 +634,23 @@ namespace swift {
     DiagnosticState(DiagnosticState &&) = default;
     DiagnosticState &operator=(DiagnosticState &&) = default;
   };
+
+  class LocalizationProducer {
+  public:
+    std::vector<DiagnosticNode> diagnostics;
+    explicit LocalizationProducer(std::string locale, std::string path);
+    
+    /// If the  message isn't available/localized in the current `yaml` file,
+    /// return the message from the `.def` file.
+    std::string getMessageOr(DiagID id, std::string defaultMessage) const;
+  };
+
+  class YAMLLocalizationProducer {
+  public:
+    LocalizationProducer localization;
```

Comment:
I thought we agreed that this is going to be a superclass of `YAMLLocalizationProducer`

Reply:
First of all I think you meant a superclass of `LocalizationProducer`. If so... If we made `YAMLLocalizationProducer` inherits from  `LocalizationProducer` I won't be able to use a constructor that initializes `localization` which is this case is a `LocalizationProducer` object. Or did you mean that `YAMLLocalizationProducer ` should use the constructor of `LocalizationProducer` like the following:
```
using LocalizationProducer::LocalizationProducer;
``

Reply:
I added this comment on `LocalizationProducer`. I expected it to be a superclass of `YAMLLocalizationProducer`.